version: '3.8'

services:
  db:
    image: mysql:8.0
    environment:
      - MYSQL_ROOT_PASSWORD=${MYSQL_PASSWORD}
      - MYSQL_DATABASE=${MYSQL_DATABASE}
    volumes:
      - db_data:/var/lib/mysql
    networks:
      - capivara_net
    ports:
      - "${MYSQL_LOCAL_PORT}:${MYSQL_DOCKER_PORT}" 
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      timeout: 20s
      retries: 10

  backend:
    build:
      context: ./api
    ports:
      - "${NODE_LOCAL_PORT}:${NODE_DOCKER_PORT}"
    networks:
      - capivara_net
    depends_on:
      db:
        condition: service_healthy
    environment:
      MYSQL_HOST: db 
      MYSQL_DATABASE: ${MYSQL_DATABASE}
      MYSQL_USERNAME: ${MYSQL_USERNAME}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}

  frontend:
    build:
      context: ./front-end
    ports:
      - "5173:80"
    depends_on:
      - backend
    networks:
      - capivara_net

networks:
  capivara_net:
    driver: bridge

volumes:
  db_data:
